src/workerpool.jl

```diff
@@ -400,3 +400,28 @@ function remotecall_pool(rc_f, f, pool::CachingPool, args...; kwargs...)
         put!(pool, worker)
     end
 end
+
+# Specialization for remotecall. We have to wait for the Future it returns
+# before putting the worker back in the pool.
+function remotecall_pool(rc_f::typeof(remotecall), f, pool::CachingPool, args...; kwargs...)
```

Comment:
Could we also add a test for it? Should be able to re-use the existing test for `WorkerPool`.

Reply:
I tried to add a test but kept hitting `CachingPool objects are not serializable.`. They seem to be missing entirely from the test suite. Perhaps that should be addressed in a follow-on PR.

Reply:
Figured one out

Reply:
Also realized these `wait`'s should be ignoring errors, as it's not the right place to throw/show them from [`4b08188` (#127)](https://github.com/JuliaLang/Distributed.jl/pull/127/commits/4b0818899a1c3a41c72a1f009ac3e7996fd6905f)